pkg/shared/service.go

```diff
@@ -18,6 +22,103 @@ func NewServiceHandler() *ServiceHandler {
 
 func (sh *ServiceHandler) Handle(ctx context.Context, event sdk.Event) error {
 	fmt.Println("handling object ", event.Object.GetObjectKind().GroupVersionKind().String())
+
+	sharedService := event.Object.(*v1alpha1.SharedService)
+	sharedServiceCopy := sharedService.DeepCopy()
+
+	if sharedService.Spec.ServiceType != strings.ToLower(v1alpha1.KeycloakKind) {
+		return nil
+	}
+
+	if event.Deleted {
+		return nil
+	}
+
+	if sharedService.GetDeletionTimestamp() != nil {
+		return sh.finalizeSharedService(sharedServiceCopy)
+	}
+
+	switch sharedService.Status.Phase {
+	case v1alpha1.SSPhaseNone:
+		sh.initSharedService(sharedServiceCopy)
+	case v1alpha1.SSPhaseAccepted:
+		sh.createKeycloaks(sharedServiceCopy)
+	case v1alpha1.SSPhaseComplete:
+		//ToDo What to do here?
+	}
+
+	return nil
+}
+
+func (sh *ServiceHandler) createKeycloaks (sharedService *v1alpha1.SharedService) error {
+	for i := 1; i <= sharedService.Spec.MinInstances; i++ {
+		err := sdk.Create(newKeycloak(sharedService))
```

Comment:
Don't we need to take a reference (e.g. name) to this created keycloak and put it into something like `SharedService.Status.Current_Instances`. Otherwise I don't see how we can find how many keycloaks we have in future for scaling this sharedservice up/down?

Reply:
Good question. I was kind of hoping/assuming since we set an owner reference on all the keycloaks that get created we could just use that when listing keycloaks for a given shared service. Maybe that's not the case though?

Reply:
oh yeah, that's cool. I had not seen that before, maybe we can do the same thing for the instance > slices relationship.